### PR TITLE
add SliceIO Implementation with specs

### PR DESCRIPTION
**Pull request by jreinert**
Two specs fail (one with segfault). I've commented them. Maybe someone can help me locate the bug?

### CODE REVIEW

File: src/io/slice_io.cr [hunk 1]
```diff
@@ -0,0 +1,75 @@
+struct SliceIO
```
Comment (jhass) [inline src/io/slice_io.cr]: I don't think this should be a struct.

**Comment by jreinert** [comment]
you're right. Not sure why it is actually...

**Comment by asterite** [comment]
This might be useful, give us some time to review it with @waj.

**Comment by jreinert** [comment]
closing in favor of MemoryIO

**Pull request closed without merging**